### PR TITLE
[WIP] Doc fix figures

### PR DESCRIPTION
**Pull request by gassmoeller**
Another step towards improving the automatic manual conversion, this time doing most of the fixes for figures. This is WIP, e.g. I did not catch all occurences of `embed` as well as `img` and I want to extract the labels as well. I will continue this soon. Currently I record the perl expressions in the commit messages, should I keep them in a script in case we need them again later?

### CODE REVIEW

**Comment by gassmoeller** [comment]
Ok, I think this is how far I can take this at  the moment. Simple figures should be converted correctly. I have seen a few instances where the original latex figures were not put in a <figure>  environment, these are still not rendering correctly. However, this should already save massive amounts of time for the manual proof-reading. Could someone review this, before we do more manual conversions?

@cmills1095 FYI, take a look at my commit messages, they detail the commands I used to make these conversions.

**Comment by tjhei** [comment]
> should I keep them in a script in case we need them again later?

yes, or at least in the doc github issue.

File: benchmarks/annulus/doc/annulus.md [hunk 1]
```diff
@@ -26,19 +26,21 @@ radial and of unit length.
 The parameter $k$ controls the number of convection cells present in the
 domain, as shown in Fig.&nbsp;[3][].
 
-<img src="cookbooks/benchmarks/annulus/doc/pressures.png" title="fig:" id="fig:annulus-vp" alt="Pressure, density and velocity fields for k=0,1,2,3 for the 2D annulus benchmark." />
-<img src="cookbooks/benchmarks/annulus/doc/density.png" title="fig:" id="fig:annulus-vp" alt="Pressure, density and velocity fields for k=0,1,2,3 for the 2D annulus benchmark." />
-<img src="cookbooks/benchmarks/annulus/doc/velocities2.png" title="fig:" id="fig:annulus-vp" alt="Pressure, density and velocity fields for k=0,1,2,3 for the 2D annulus benchmark." />
+<img src="pressures.png" title="fig:" id="fig:annulus-vp" alt="Figure" />
```
Comment (tjhei) [inline benchmarks/annulus/doc/annulus.md:29]: can you turn the "alt" text into a caption?

File: benchmarks/buiter_et_al_2016_jsg/doc/buiter_et_al_2016_jsg.md [hunk 2]
```diff
@@ -13,8 +13,8 @@ differently-colored sand and compresses or stretches the volume. Examples of
 the patterns one can then observe in these do-it-yourself models are shown in
 Fig.&nbsp;[2][].
 
-<img src="cookbooks/benchmarks/buiter_et_al_2016_jsg/doc/real-sandbox-1.jpg" title="fig:" id="fig:sandbox-images" style="width:45.0%" alt="Examples of deformation patterns of &#x201C;sand box&#x201D; experiments in which alternating layers of differently-colored sand undergo deformation. Pictures courtesy of the lab of Dennis Harry at Colorado State University." />
-<img src="cookbooks/benchmarks/buiter_et_al_2016_jsg/doc/real-sandbox-2.jpg" title="fig:" id="fig:sandbox-images" style="width:45.0%" alt="Examples of deformation patterns of &#x201C;sand box&#x201D; experiments in which alternating layers of differently-colored sand undergo deformation. Pictures courtesy of the lab of Dennis Harry at Colorado State University." />
+<img src="real-sandbox-1.jpg" title="fig:" id="fig:sandbox-images" style="width:45.0%" alt="Figure" />
+<img src="real-sandbox-2.jpg" title="fig:" id="fig:sandbox-images" style="width:45.0%" alt="Figure" />
```
Comment (tjhei) [inline benchmarks/buiter_et_al_2016_jsg/doc/buiter_et_al_2016_jsg.md:17]: ``alt=`` is pretty useless here...

File: benchmarks/crameri_et_al/doc/crameri_et_al.md [hunk 3]
```diff
@@ -15,9 +15,11 @@ are in Figure&nbsp;[1][].
...
-</figure>
+```{figure-md} fig:crameri-benchmark-initial-topography
+<img src="initial_topography.png" style="width:95.0%" alt="Figure" />
+
+Setup for the topography relaxation benchmark. The box is <span class="math inline">2800</span> km wide and <span class="math inline">700</span> km high, with a <span class="math inline">100</span> km lid on top. The lid has a viscosity of <span class="math inline">10<sup>23</sup>&#x2006;<em>P</em><em>a</em>&#x2006;<em>s</em></span>, while the mantle has a viscosity of <span class="math inline">10<sup>21</sup>&#x2006;<em>P</em><em>a</em>&#x2006;<em>s</em></span>. The sides are free slip, the bottom is no slip, and the top is a free surface. Both the lid and the mantle have a density of <span class="math inline">3300&#x2006;<em>k</em><em>g</em>/<em>m</em><sup>3</sup></span>, and gravity is <span class="math inline">10&#x2006;<em>m</em>/<em>s</em><sup>2</sup></span>. There is a <span class="math inline">7&#x2006;<em>k</em><em>m</em></span> sinusoidal initial topography on the free surface.
```
Comment (tjhei) [inline benchmarks/crameri_et_al/doc/crameri_et_al.md:21]: this will need some manual work.

File: benchmarks/gravity_mantle/doc/gravity_mantle.md [hunk 4]
```diff
@@ -72,8 +72,8 @@ contribution to the gravity signal of any density distribution that solely
 depends on $r$ can and should be removed as it does not contain any valuable
 information.
 
-<img src="cookbooks/benchmarks/gravity_mantle/doc/grav.*" title="fig:" id="fig:grav_mantle2" style="width:48.0%" alt="Mantle gravity: gravitational acceleration |g| (left) and potential (right) computed at radius 6621&#xA0;km." />
-<img src="cookbooks/benchmarks/gravity_mantle/doc/pot.*" title="fig:" id="fig:grav_mantle2" style="width:48.0%" alt="Mantle gravity: gravitational acceleration |g| (left) and potential (right) computed at radius 6621&#xA0;km." />
+<img src="cookbooks/benchmarks/gravity_mantle/doc/grav.*" title="fig:" id="fig:grav_mantle2" style="width:48.0%" alt="Figure" />
+<img src="cookbooks/benchmarks/gravity_mantle/doc/pot.*" title="fig:" id="fig:grav_mantle2" style="width:48.0%" alt="Figure" />
```
Comment (tjhei) [inline benchmarks/gravity_mantle/doc/gravity_mantle.md:76]: again, no reason to keep ``alt=`` and the caption is gone

File: benchmarks/slab_detachment/doc/slab_detachment.md [hunk 5]
```diff
@@ -15,9 +15,11 @@ computational domain is a $1000 \si{km}\times 660 \si{km}$ box. No-slip
 boundary conditions are imposed on the sides of the system, while free-slip
 boundary conditions are imposed at the top and bottom.
 
-<figure>
-<img src="cookbooks/benchmarks/slab_detachment/doc/drawing.png" id="fig:slab_detachment_setup" alt="Slab detachment benchmark: Initial geometry [fig:slab_detachment_setup]" /><figcaption aria-hidden="true"><em>Slab detachment benchmark: Initial geometry <span id="fig:slab_detachment_setup" label="fig:slab_detachment_setup">[fig:slab_detachment_setup]</span></em></figcaption>
-</figure>
+```{figure-md} fig:slab_detachment_setup
+<img src="drawing.png" alt="Figure" />
+
+Slab detachment benchmark: Initial geometry <span id="fig:slab_detachment_setup" label="fig:slab_detachment_setup">[fig:slab_detachment_setup]</span>
```
Comment (tjhei) [inline benchmarks/slab_detachment/doc/slab_detachment.md:21]: not sure what that span is about...

**Comment by gassmoeller** [comment]
Superseded by #4832

**Pull request closed without merging**